app/src/main/java/com/example/crosses_zeros/ui/presentation/gameboard/GameBoardViewModel.kt

```diff
@@ -0,0 +1,192 @@
+        when {
+            firstRow() || secondRow() || thirdRow() ||
+                    firstColumn() || secondColumn() || thirdColumn() ||
+                    firstDiagonal() || secondDiagonal() -> {
+                viewModelScope.launch {
+                    gameEnd.emit(true)
+                }
+                toDrawWinLine()
+            }
+        }
+    }
+
+    private fun setListData(first: Int, second: Int, third: Int) {
+        crossedOutCells.update {
+            val list = mutableListOf(first, second, third)
+            list.toList()
+        }
+    }
+
+    private fun toDrawWinLine() {
+        when (gameEnd.value) {
+            firstRow() -> setListData(0, 1, 2)
+            secondRow() -> setListData(3, 4, 5)
+            thirdRow() -> setListData(6, 7, 8)
+            firstColumn() -> setListData(0, 3, 6)
+            secondColumn() -> setListData(1, 4, 7)
+            thirdColumn() -> setListData(2, 5, 8)
+            firstDiagonal() -> setListData(0, 4, 8)
+            else -> setListData(2, 4, 6)
+        }
+    }
+
+    private fun secondDiagonal() =
+        mutableGameBoardState.value[2] == element &&
+                mutableGameBoardState.value[4] == element &&
+                mutableGameBoardState.value[6] == element
+
+    private fun firstDiagonal() =
+        mutableGameBoardState.value[0] == element &&
+                mutableGameBoardState.value[4] == element &&
+                mutableGameBoardState.value[8] == element
+
+    private fun thirdColumn() =
+        mutableGameBoardState.value[2] == element &&
+                mutableGameBoardState.value[5] == element &&
+                mutableGameBoardState.value[8] == element
+
+    private fun secondColumn() =
+        mutableGameBoardState.value[1] == element &&
+                mutableGameBoardState.value[4] == element &&
+                mutableGameBoardState.value[7] == element
+
+    private fun firstColumn() =
+        mutableGameBoardState.value[0] == element &&
+                mutableGameBoardState.value[3] == element &&
+                mutableGameBoardState.value[6] == element
+
+    private fun thirdRow() =
+        mutableGameBoardState.value[6] == element &&
+                mutableGameBoardState.value[7] == element &&
+                mutableGameBoardState.value[8] == element
+
+    private fun secondRow() =
+        mutableGameBoardState.value[3] == element &&
+                mutableGameBoardState.value[4] == element &&
+                mutableGameBoardState.value[5] == element
+
+    private fun firstRow() =
+        mutableGameBoardState.value[0] == element &&
+                mutableGameBoardState.value[1] == element &&
+                mutableGameBoardState.value[2] == element
+
+    fun onStartNewGame() {
+        mutableGameBoardState.update {
+            val list = it.toMutableList()
+            list.clear()
+            list.addAll(emptyList)
+            list.toList()
+        }
+        crossedOutCells.update {
```

Comment:
```suggestion
        crossedOutCells.update { listOf() }
```

Reply:
Here the list is not empty, it includes eight empty pictures.

Reply:
Here the list is not empty, it includes eight empty pictures.

Reply:
It was replace